src/utils.ts

```diff
@@ -1319,6 +1321,23 @@ export class HostAddress {
     Object.freeze(this);
   }
 
+  equals(other: HostAddress): boolean {
+    return (
```

Comment:
just double checking that we definitely want to be strict even in cases of null/undefined with the triple equals here